lib/praxis/api_definition.rb

```diff
@@ -41,35 +46,45 @@ def trait(name, &block)
       if self.traits.has_key? name
         raise Exceptions::InvalidTrait.new("Overwriting a previous trait with the same name (#{name})")
       end
-      self.traits[name] = block
+      self.traits[name] = Trait.new(&block)
     end
 
-    # Setting info to the nil version, means setting it for all versions (if they don't  override them)
-    def info( version=nil, &block)
-      i = @infos[version]
-      i.instance_eval(&block)
-      i
+    # Setting info to the nil version, means setting it for all versions (if they don't override them)
+    def info(version=nil, &block)
+      if version.nil?
+        if block_given?
+          @global_info.instance_eval(&block)
+        else
+          @global_info
+        end
+      else
+        i = @infos[version]
+        if block_given?
+          i.instance_eval(&block)
+        end
+        i
+      end
     end
-    
+
     def describe
-      global_info = @infos[nil].describe
       data = Hash.new do |hash, version|
         hash[version] = Hash.new
       end
       # Fill in the "info" portion
       @infos.each do |version,info|
-        next unless version
-        info_hash = global_info.merge(info.describe)
-        [:name, :title].each do |attr|
-          raise "Error: API Global information for version '#{version}' does not have '#{attr}' defined. " unless info_hash.key? attr
+        data[version][:info] = info.describe
+      end
+
```

Comment:
I think we also need to add a "global" or "all_versions" type key where we dump the value of the `@global_info` instance. I.e., some info that applies to all versions (i.e. which could mostly be about text and description...etc...) which would be shown at the very entrance of the API docs...